### ISSUE TITLE
Auto resize broken since 2.0.2

### ISSUE THREAD

**Issue by lwis** [BTMorton/angular2-grid#249]
The grid remains the same size as the initial viewport, resizing the viewport doesn't change the size after 2.0.2.

**Comment by scrodrigues** [comment]
Hi, 

I'm having the same issue. Any solution?

best

**Comment by Ketec** [comment]
For now  the window resize i could hotfix with:
```
    @ViewChild(NgGrid) public ngGrid: NgGrid;

    constructor() {
        window.onresize = (e) => {
            this.ngGrid.resizeEventHandler(e);
        };
    }
```

But it doesn't help with any non window specific container resizes in a dynamic layout. Plus items do not smoothly move when resizing/get pushed - instead instantly reset to default 0x0 location - seemingly ignoring col/row values.

**Comment by Ketec** [comment]
not 100% sure but perhaps because of this:
`this._renderer.setElementStyle(this._ngEl.nativeElement, 'width', '100%');//(maxCol * (this.colWidth + this.marginLeft + this.marginRight))+'px');`
https://github.com/BTMorton/angular2-grid/blob/master/src/directives/NgGrid.ts#L1114

**Comment by BTMorton** [comment]
Can you create a plunker to demonstrate this?

**Comment by sconix** [comment]
We have been effected by this as well. Previously we were able to go around this by calling resizeEventHandler on ngAfterViewInit, but not anymore (don't remember when this workaround got broken and was it even needed with 2.0.2.

I can try to find some time during this week to look into this bit more and provide more info or even PR.

**Comment by sconix** [comment]
I started to look into this and of course now it does not happen :) I believe that latest flex-layout library release fixed it so at the moment I don't have way to reproduce this. If it gets broken again I will look into this more, but now its working nicely.

**Comment by lwis** [comment]
This still impacts me, I'm just not sure how to create a plunkr to demonstrate it.

My config looks like this;

```
private config = {
        'margins': [0],             //  The size of the margins of each item. Supports up to four values in the same way as CSS margins. Can be updated using setMargins()
        'draggable': false,         //  Whether the items can be dragged. Can be updated using enableDrag()/disableDrag()
        'resizable': false,         //  Whether the items can be resized. Can be updated using enableResize()/disableResize()
        'max_cols': 10,              //  The maximum number of columns allowed. Set to 0 for infinite. Cannot be used with max_rows
        // 'max_rows': 10,             //  The maximum number of rows allowed. Set to 0 for infinite. Cannot be used with max_cols
        // 'visible_cols': 10,          //  The number of columns shown on screen when auto_resize is set to true. Set to 0 to not auto_resize. Will be overriden by max_cols
        'visible_rows': 10,          //  The number of rows shown on screen when auto_resize is set to true. Set to 0 to not auto_resize. Will be overriden by max_rows
        'min_cols': 1,              //  The minimum number of columns allowed. Can be any number greater than or equal to 1.
        'min_rows': 1,              //  The minimum number of rows allowed. Can be any number greater than or equal to 1.
        // 'col_width': 192,           //  The width of each column
        // 'row_height': 108,           //  The height of each row
        'cascade': 'left',            //  The direction to cascade grid items ('up', 'right', 'down', 'left')
        // 'min_width': 192,           //  The minimum width of an item. If greater than col_width, this will update the value of min_cols
        // 'min_height': 108,           //  The minimum height of an item. If greater than row_height, this will update the value of min_rows
        'fix_to_grid': true,       //  Fix all item movements to the grid
        'auto_style': true,         //  Automatically add required element styles at run-time
        'auto_resize': true,        //  Automatically set col_width/row_height so that max_cols/max_rows fills the screen. Only has effect is max_cols or max_rows is set
        'maintain_ratio': true,    //  Attempts to maintain aspect ratio based on the colWidth/rowHeight values set in the config
        'prefer_new': false,        //  When adding new items, will use that items position ahead of existing items
        'limit_to_screen': true,   //  When resizing the screen, with this true and auto_resize false, the grid will re-arrange to fit the screen size. Please note, at present this only works with cascade direction up.
    };
```

**Comment by sconix** [comment]
Actually the bug is still there I just forgot that I downgraded to 2.0.0 while a go to get the resize working :) I will look into this tomorrow to see if I can find the cause.

**Comment by sconix** [comment]
I looked into this and it seems that to get the old behavior would be to have both auto_resize and limit_to_screen set to true, but as the documentation says thats not supported. 

@BTMorton why that combination is not supported anymore? Thats how we need to have the grid, i.e. auto_resize the blocks but also do that when resizing screen. Thats how auto_resize alone worked before 2.0.2. Would you accept a PR that allows using both options to get the behavior that 2.0.0 had?

**Comment by sconix** [comment]
Actually with bit more thinking / testing the way limit_to_screen works is better for our purposes, but the only problem is that we would need to be able to center the grid since with fixed col size it can be that there is unusable space on the right.

I think this would be easily solved by having center_to_screen etc. option that would apply some extra margin based on grid width and max col count. Or we could do this in application but even for that we would need a way to get the maxCols size on resize.

I can make a PR, but which one would be better? To have an option for centering the grid contents or to expose the maxCols via event.

**Comment by BTMorton** [comment]
`auto_resize` updates the column width and row heights dynamically based upon a `max_cols` or `max_rows` value. This is not supported with `limit_to_screen` because it would make the functionality redundant as the items would never overflow the screen in the first place.

If this is just a case of the column/row width resizing as the window resizes, that's should hopefully be a relatively simple fix. In fact, I'm pretty sure I know what the issue is now and it shouldn't take too long to fix.

**Comment by sconix** [comment]
Yes I do now understand how those two options should be working, the problem just is that when using just auto_resize nothing is done on window resize so hence the layout wont update. The items are only adjusted if limit_to_screen is used.

**Comment by sconix** [comment]
I was about to fix it and submit PR on monday, but great if you can look into it. I am also going to do the center_to_screen option that can be used with the limit_to_screen to center the grid. We need it in our dashboard and I already did a quick implementation of it, but will do better which wont need re-positioning of all items on resize.

**Comment by BTMorton** [comment]
This has been fixed in v2.0.7

**Comment by lwis** [comment]
Confirmed, thanks @BTMorton
